### PR TITLE
fix SegmentedControl for compat preact

### PR DESCRIPTION
**Pull request by pingan1927**
ref #1304 

First of all, thanks for your contribution! :-)

Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1374)

### CODE REVIEW

File: components/segmented-control/index.web.tsx [hunk 1]
```diff
@@ -56,11 +56,21 @@ export default class SegmentedControl extends React.Component<SegmentedControlPr
       [`${prefixCls}-item-selected`]: selected,
     });
 
-    const itemStyle = {
-      color: selected ? '#fff' : tintColor,
-      backgroundColor: selected ? tintColor : '#fff',
-      borderColor: tintColor,
-    };
```
Comment (pingan1927) [inline components/segmented-control/index.web.tsx]: `SegmentedControl`的渲染问题 在preact-compat层面无法修改
原因在于react对style对象做了很多异常值处理
而preact就简单的赋值而已
所以兼容preact需要把`SegmentedControl`关于style对象的代码写的更健壮些 改动不大。

Comment (silentcloud) [inline components/segmented-control/index.web.tsx]: @pingan1927 这样改看着好累，defaultProps 里给 tintColor 传一个空字符串，应该也可以解决问题吧

**Comment by silentcloud** [comment]
@pingan1927 ci

File: components/segmented-control/__tests__/__snapshots__/demo.test.web.js.snap [hunk 2]
```diff
@@ -18,7 +18,7 @@ exports[`renders ./components/segmented-control/demo/basic.md correctly 1`] = `
       aria-selected="true"
       class="am-segment-item am-segment-item-selected"
       role="tab"
-      style="color:#fff;"
+      style="color:#fff;background-color:;border-color:;"
```
Comment (paranoidjk) [inline components/segmented-control/__tests__/__snapshots__/demo.test.web.js.snap]: @silentcloud @pingan1927 这个style真的没问题？

Comment (pingan1927) [inline components/segmented-control/__tests__/__snapshots__/demo.test.web.js.snap]: 刚又试了试，没有问题

Comment (silentcloud) [inline components/segmented-control/__tests__/__snapshots__/demo.test.web.js.snap]: @paranoidjk 没问题的，空 style 可以的

**Pull request merged**